### ISSUE TITLE
Warning: Missing argument 1 for CrudKit\Pages\BasicLoginPage::preprocess()

### ISSUE THREAD

**Issue by stef-code** [skyronic/crudkit#54]
When visiting the page, I get this PHP warning:

> Warning: Missing argument 1 for CrudKit\Pages\BasicLoginPage::preprocess(), called in \vendor\skyronic\crudkit\src\CrudKit\CrudKitApp.php on line 51 and defined in \vendor\skyronic\crudkit\src\CrudKit\Pages\BasicLoginPage.php on line 67

In CrudKitApp.php, there is this function call of preprocess() without arguments, see line 51:
```
48     protected $login = null;
49     public function useLogin ($login) {
50         $this->login = $login;
51         $login->preprocess ();
52         if (!$login->check ()) {
53             $this->addPage ($login);
54             $this->render ();
55         }
56     }
```
File BasicLoginPage.php has got a function preprocess, but takes one argument $app, see line 67:
```
67 	public function preprocess ($app) {
68 		if ($this->url->get ('__ckLogout', false) !== false) {
69 			$this->doLogout ();
70 		}
71 	}
```

**Comment by fncapps** [comment]
Same here.

It shows up when going into the login page.

**Comment by mariox92** [comment]
I deleted `$app` argument in BasicLoginPage.php as it seems that it is not used at all

**Comment by stef-code** [comment]
Yes, also using something like `$login->preprocess ($this);` works as a workaround.

**Comment by genofire** [comment]
:+1:

**Comment by jaeturma** [comment]
I encounter the same error.. which one should work well?

**Comment by jamespsterling** [comment]
I removed the $app argument from the /vendor/skyronic/crudkit/src/CrudKit/Pages/BasicLoginPage.php, and the page rendered properly. Seems like a bug.
